searchlib/src/vespa/searchlib/attribute/attributevector.cpp

```diff
@@ -788,11 +788,16 @@ void
 AttributeVector::update_config(const Config& cfg)
 {
     commit(true);
-    drain_hold(1024 * 1024); // Wait until 1MiB or less on hold
+    bool compaction_strategy_changed = (cfg.getCompactionStrategy() != _config.getCompactionStrategy());
```

Comment:
I think it would be easier to read if you moved _config.setGrowStrategy(cfg.getGrowStrategy()) first, and then added a guard if (cfg.getCompactionStrategy() == _config.getCompactionStrategy()) return;
Then you do not need the 2 ifs.